### DISCUSSION TITLE
OSCAL extension - mappings

### DISCUSSION THREAD

**Discussion by bjcbusiness** [oscal-compass/compliance-trestle#1615] [Ideas]
OSCAL has defined a schema for declaring mappings between elements. Has there been any thought about how (and whether) to implement this in trestle?

Reference: https://pages.nist.gov/OSCAL-Reference/models/prototype-mapping-model/mapping/

**Comment by degenaro** [comment]
Thanks for this timely inquiry! The reference you cite is a work-in-progress. 

A bit of history. The mapping model was proposed perhaps 2 or more years ago, and a draft model was created by the NIST OSCAL team slightly different from the current proposal, but not drastically different. There is an old trestle branch that supports the older OSCAL 1.0.2. models plus the old draft mapping model. Just as a coincidence, I created another branch this morning that supports OSCAL 1.1.2 in a hackish kind of way plus the old draft mapping model. What I mean by this is that the version check code has been hacked to accept 1.1.2 without actually code to provide such support, so caveat emptor. All should work well, if you are just being backward compatible with the other models. Using new features (beyond 1.0.2) would presumably not work. That branch is here: [https://github.com/oscal-compass/compliance-trestle/tree/bridge/synthesize-cd](https://github.com/oscal-compass/compliance-trestle/tree/bridge/synthesize-cd).
All of this is not officially supported by trestle, but it does work and minimal effort to fix things if necessary would be entertained.

With luck, the new mapping model will be presented by the NIST OSCAL team to the community in the near future. Whether trestle will try to adopt a pre-release version of the mapping model or wait for official release is TBD (my guess is the latter). You are welcome to attend the regular CNCF oscal-compass community meeting to discuss. See [https://github.com/oscal-compass](https://github.com/oscal-compass).

**Comment by bjcbusiness** [comment]
Hello,

That's great news! Thank you for working on that. Later on I'll take a look and see whether I can manage. I'm completely new to compliance-trestle and figure I should get comfortable with the released version, then take a closer look at the "bleeding edge" stuff.

The mappings extension is very important, I think-at least based on the number of mapping spreadsheet tables that I'm sure we've all written or used.

Regards
Ben

**Comment by nickmahl** [comment]
Hi,

I am also interested in trying the OSCAL Mapping feature even though it is still in the prototyping stage. I wanted to ask whether the branches you previously mentioned @degenaro are still the recommended way to experiment with it, or if there are any newer branches available now. Thanks in advance.

**Comment by degenaro** [comment]
Nothing newer yet, but we are getting closer. See https://matrix.to/#/!KiqEhMrfjXqEyIwUMw:gitter.im/$0FU5AWAduqPFJ7k6d1VLAT4UJ8gPwGs3GkwsPCL2dew?via=gitter.im&via=matrix.org&via=matrix.freyachat.eu and https://oscalfoundation.slack.com/archives/C09HGN152KZ/p1763937648273539.

**Reply by nickmahl** [reply]
Thanks for the info, very interesting indeed :) I implemented the Mapping Collection myself now in a fork for testing. I also added a script that resolves all controls from other catalogs that have a (transitive) mapping relationship. This is already very useful now. Finally, no Excel Mapping Sheets anymore.
